components/nexus-ehcache/src/main/java/org/sonatype/nexus/cache/EhCacheManagerProvider.java

```diff
@@ -43,24 +49,48 @@
     extends ComponentSupport
     implements Provider<CacheManager>
 {
-  private final URI uri;
+  private static final String CONFIG_FILE = "ehcache.xml";
+
+  private static final String DEFAULT_RESOURCE = "ehcache-default.xml";
+
+  private volatile CacheManager cacheManager;
 
-  private CacheManager cacheManager;
+  // TODO: Consider if we want to lazily create the instance on first get?  Unsure if it matters or not though.
 
   @Inject
   public EhCacheManagerProvider(final ApplicationDirectories directories) {
-    File file = new File(directories.getAppDirectory("etc"), "ehcache.xml");
-    checkState(file.exists(), "Missing required file: %s", file.getAbsolutePath());
-    this.uri = file.toURI();
+    checkNotNull(directories);
+
+    URI uri = null;
+    File file = new File(directories.getAppDirectory("etc"), CONFIG_FILE);
+    if (file.exists()) {
+      uri = file.toURI();
+    }
+    else {
+      log.warn("Missing configuration: {}", file.getAbsolutePath());
+    }
+    this.cacheManager = create(uri);
   }
 
   @VisibleForTesting
-  public EhCacheManagerProvider(final URI uri) {
-    this.uri = checkNotNull(uri);
+  public EhCacheManagerProvider(@Nullable final URI uri) {
+    this.cacheManager = create(uri);
   }
 
-  @VisibleForTesting
-  synchronized CacheManager create() {
+  private CacheManager create(@Nullable URI uri) {
+    if (uri == null) {
+      // load the configuration from defaults, this is mainly used for test environments
+      log.warn("Using default configuration");
+      URL url = getClass().getResource(DEFAULT_RESOURCE);
+      checkState(url != null, "Missing configuration resource: %s", DEFAULT_RESOURCE);
+      try {
+        uri = url.toURI();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "uri". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)